pkgs/applications/editors/vim/plugins/overrides.nix

```
@@ -1532,6 +1532,10 @@ in
     dependencies = with self; [ ultisnips ];
   };
 
+  neoconf-nvim = super.neoconf-nvim.overrideAttrs {
+    dependencies = with self; [ nvim-lspconfig ];
```

It's a good practice to also add a smoke test

```suggestion
    dependencies = with self; [ nvim-lspconfig ];

    doInstallCheck = true;
    nvimRequireCheck = "neoconf";
```